### PR TITLE
MIM-11: Add currency config for EUR

### PR DESCRIPTION
**Pull request by corydeppen**
- Adds currency config for EUR
- Adds ESLint and Prettier
- Updates deprecated Rollup config options

### CODE REVIEW

File: lib/currency/index.js [hunk 1]
```diff
@@ -32,41 +46,44 @@ var symbols = {
     format: '$0,0.00',
     raw: '0,0.00'
   }
-};
+}; // eslint-disable-next-line no-unused-vars
 
 function parseLocaleNumber(stringNumber) {
   var thousandSeparator = 1111 .toLocaleString().replace(/1/g, '');
```
Comment (CenturionD) [inline lib/currency/index.js]: Is that a space between 1111 and toLocaleString()?

File: lib/currency/index.js [hunk 2]
```diff
@@ -32,41 +46,44 @@ var symbols = {
...
   var decimalSeparator = 1.1.toLocaleString().replace(/1/g, '');
-  return parseFloat(stringNumber.replace(new RegExp('\\' + thousandSeparator, 'g'), '').replace(new RegExp('\\' + decimalSeparator), '.'));
-}
+  return parseFloat(stringNumber.replace(new RegExp("\\".concat(thousandSeparator), 'g'), '').replace(new RegExp("\\".concat(decimalSeparator)), '.'));
+} // eslint-disable-next-line import/prefer-default-export
+
 
 function setLocale() {
   var locale = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'USD';
```
Comment (CenturionD) [inline lib/currency/index.js]: Wow is this hokey or what?

This isn't you @corydeppen but I hate this whole f1-utils library. We should've just used a library that did this and some template strings.

**Comment by CenturionD** [comment]
Another question, how do we release this update?

File: lib/currency/index.js (hunk 1 above)
Comment (corydeppen) [reply lib/currency/index.js]: Looks like it - I'm unsure why Babel needed to remove the parens like that ([repl example](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABABwIYCcDOBTAMnCVAG2wDkQBbAI23QApMp0YwBzc62gSkQG8AoRIggJGiKAAs4ITKjAATAMrY06VFDjpEAXkR0AjIf1cAdBvyESipi1Z1T6FUVQRsdAPT73rADSIA5P5cgsKiUIjy2BAwFMTKquqaOnr6JsZmcBbE2NbMbPYmjsjOrh5evgFB_CGOUCDoSKo4AGJEcOp0IUKMeeyUNOhdQoiFTi5uYNgA7ogAStisAKIAHsh0AAYAOpsAJLyS0rIK8RiJ6AC-637-rEHXVcPDo8XjdJMz80urG9t7kdGxIgnNQaC7rLjXEwPRDBc5AA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.7.3&externalPlugins=)).

It looks like that function may have only been used to test something and can be removed at this point. Thoughts?

**Comment by corydeppen** [comment]
> Another question, how do we release this update?

Portal and mono both use a Git URL that includes a tag, so I think we'd just merge the PR and tag the release. The package isn't published anywhere as far as I know.

**Comment by corydeppen** [comment]
> I hope one day we get rid of this library in favor of some well-supported npm package that does this formatting for us instead of supporting our own homegrown thing.

I like the spirit of this library, as I'd imagine there are several utility functions that can be shared across portal and services. For currency formatting specifically, I think the approach was to enhance the Numeral library to bake in the formatting to and from cents, as well as indicate the symbol position for use on input fields in the portal.

I do have concerns about how we're using the Numeral library, though. We're taking the currency code as an arg to the `setLocale` function, but this differs from the intent of the Numeral library that expects a language tag. It also means we're not taking advatange of the Numeral locale files. Since the currency formatting can differ by locale, I think we'd want to pass the user's locale and not the currency code defined for the application. Of course, this assumes the possibility of multiple users with different locales using a given white label's app.

To make things a bit more "interesting", it seems the Numeral library may be on life support after reading a few issues in the [Numeral repo](https://github.com/adamwdraper/Numeral-js) (e.g. 638, 431, and 242). Considering more than just download figures for the two, we may want to investigate [numbro](https://github.com/BenjaminVanRyseghem/numbro) that's mentioned in those issues due to what appears to be more active development and support (i.e. 110 open PRs in the Numeral repo and a maintainer that doesn't seem involved in conversations around the health of his library).

Thoughts?

**Comment by CenturionD** [comment]
> > I hope one day we get rid of this library in favor of some well-supported npm package that does this formatting for us instead of supporting our own homegrown thing.
> 
> I like the spirit of this library, as I'd imagine there are several utility functions that can be shared across portal and services. For currency formatting specifically, I think the approach was to enhance the Numeral library to bake in the formatting to and from cents, as well as indicate the symbol position for use on input fields in the portal.
> 
> I do have concerns about how we're using the Numeral library, though. We're taking the currency code as an arg to the `setLocale` function, but this differs from the intent of the Numeral library that expects a language tag. It also means we're not taking advatange of the Numeral locale files. Since the currency formatting can differ by locale, I think we'd want to pass the user's locale and not the currency code defined for the application. Of course, this assumes the possibility of multiple users with different locales using a given white label's app.
> 
> To make things a bit more "interesting", it seems the Numeral library may be on life support after reading a few issues in the [Numeral repo](https://github.com/adamwdraper/Numeral-js) (e.g. 638, 431, and 242). Considering more than just download figures for the two, we may want to investigate [numbro](https://github.com/BenjaminVanRyseghem/numbro) that's mentioned in those issues due to what appears to be more active development and support (i.e. 110 open PRs in the Numeral repo and a maintainer that doesn't seem involved in conversations around the health of his library).
> 
> Thoughts?

I have to say that the rest of the engineering team explained from the inception of this library was a bad idea since we needed to maintain our own currency/address formatting functions when libraries that specialize in both separately already exist and are better supported. When we do finally make server fully i18n, we should not carry this f1-utils library with us in favor of currency/address npm packages that must exist somewhere.

https://www.npmjs.com/package/dinero.js
Above is something that looks like a good alternative that includes string formats and it took like around 10 mins of searching to find one that had what we needed (requesting any currency number, doing simple math operations with number parsing, and output with a specific string format). Unless there's some other functionality needed that HAS to be shared in this one repo, we could've used address formatting/parsing from another library that absolutely exists in npm somewhere.

**Pull request merged**